### PR TITLE
Refs #29214: Only install Pulp and Qpid on El7

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Is there a clean way to include parameters based on the OS version only? While failing is nice, in the installer use case I wouldn't want any of these showing up to the users.

**Comment by ekohl** [comment]
No, but those options have been available for a long time and users don't appear to be using them that much anyway. I'd propose we fix make sure it works correctly, then look at trimming down options. We'll drop Pulp 2 soon anway.

File: manifests/init.pp [hunk 1]
```diff
@@ -214,7 +218,7 @@
     }
   }
 
-  if $pulp_master or $pulp {
+  if ($pulp_master or $pulp) and $facts['os']['release']['major'] == '7' {
```
Comment (ekohl) [inline manifests/init.pp:221]: All these checks are now redundant since we fail early on.

Comment (ehelms) [reply manifests/init.pp:221]: Safety first!

Comment (ehelms) [reply manifests/init.pp:221]: That is me saying, the fail does cover it but I also see this has helping to cover our bases and make it clear in the code this block is EL7 only. I debated moving a bunch of htis to their own class to highlight that as well but didn't want to do too much re-factoring.

Comment (ekohl) [reply manifests/init.pp:221]: Feels a bit redundant to me since you know there's no Pulp 2 on EL8 anyway.

Comment (ehelms) [reply manifests/init.pp:221]: I know that, but doesn't mean everyone reading this code will clearly know that.

Comment (ekohl) [reply manifests/init.pp]: Partly I agree, but the code is already very hard to follow. Adding more conditionals just makes it harder rather than easier IMHO. I'm waiting till we can drop Pulp for a big refactor.
